src/cider/tests/CiderNewAggTest.cpp

```diff
@@ -0,0 +1,285 @@
+  reinterpret_cast<uint8_t*>(key2_ptr + 2)[0] = (uint8_t)2;
+  // Use get api and return value address
+  int8_t* value2_ptr = aggregator.get(key2_ptr);
+
+  // Check init value
+  CHECK_EQ(reinterpret_cast<int64_t*>(value2_ptr)[0], 0);
+  CHECK_EQ(reinterpret_cast<int32_t*>(value2_ptr + 8)[0], 0);
+  CHECK_EQ(reinterpret_cast<int8_t*>(value2_ptr + 12)[0],
+           std::numeric_limits<int8_t>::max());
+  CHECK_EQ(reinterpret_cast<int8_t*>(value2_ptr + 13)[0],
+           std::numeric_limits<int8_t>::min());
+
+  // Some agg operations and update the value, like SUM, COUNT, MIN, MAX
+  reinterpret_cast<int64_t*>(value2_ptr)[0] += 20;
+  reinterpret_cast<int32_t*>(value2_ptr + 8)[0] += 1;
+  reinterpret_cast<int8_t*>(value2_ptr + 12)[0] =
+      std::min(reinterpret_cast<int8_t*>(value2_ptr + 12)[0], (int8_t)20);
+  reinterpret_cast<int8_t*>(value2_ptr + 13)[0] =
+      std::max(reinterpret_cast<int8_t*>(value2_ptr + 13)[0], (int8_t)20);
+
+  // Row2:
+  int8_t* key3_ptr = allocator->allocate(4);
+  reinterpret_cast<bool*>(key3_ptr)[0] = (bool)false;
+  reinterpret_cast<uint16_t*>(key3_ptr + 2)[0] = (uint8_t)1;
+  // Use get api and return value address
+  int8_t* value3_ptr = aggregator.get(key3_ptr);
+
+  // Some agg operations and update the value, like SUM, COUNT, MIN, MAX
+  reinterpret_cast<int64_t*>(value3_ptr)[0] += 30;
+  reinterpret_cast<int32_t*>(value3_ptr + 8)[0] += 1;
+  reinterpret_cast<int8_t*>(value3_ptr + 12)[0] =
+      std::min(reinterpret_cast<int8_t*>(value3_ptr + 12)[0], (int8_t)30);
+  reinterpret_cast<int8_t*>(value3_ptr + 13)[0] =
+      std::max(reinterpret_cast<int8_t*>(value3_ptr + 13)[0], (int8_t)30);
+
+  // Final check
+  // Check key = 1
+  int8_t* key1_check_ptr = allocator->allocate(4);
+  reinterpret_cast<bool*>(key1_check_ptr)[0] = (bool)false;
+  reinterpret_cast<uint8_t*>(key1_check_ptr + 2)[0] = (uint8_t)1;
+  // Use get api and return value address
+  int8_t* value1_check_ptr = aggregator.get(key1_check_ptr);
+
+  // Check agg result value
+  CHECK_EQ(reinterpret_cast<int64_t*>(value1_check_ptr)[0], 40);
+  CHECK_EQ(reinterpret_cast<int32_t*>(value1_check_ptr + 8)[0], 2);
+  CHECK_EQ(reinterpret_cast<int8_t*>(value1_check_ptr + 12)[0], 10);
+  CHECK_EQ(reinterpret_cast<int8_t*>(value1_check_ptr + 13)[0], 30);
+
+  // Check key = 2
+  int8_t* key2_check_ptr = allocator->allocate(4);
+  reinterpret_cast<bool*>(key2_check_ptr)[0] = (bool)false;
+  reinterpret_cast<uint8_t*>(key2_check_ptr + 2)[0] = (uint8_t)2;
+  // Use get api and return value address
+  int8_t* value2_check_ptr = aggregator.get(key2_check_ptr);
+
+  // Check agg result value
+  CHECK_EQ(reinterpret_cast<int64_t*>(value2_check_ptr)[0], 20);
+  CHECK_EQ(reinterpret_cast<int32_t*>(value2_check_ptr + 8)[0], 1);
+  CHECK_EQ(reinterpret_cast<int8_t*>(value2_check_ptr + 12)[0], 20);
+  CHECK_EQ(reinterpret_cast<int8_t*>(value2_check_ptr + 13)[0], 20);
+}
+
+TEST_F(CiderNewAggTest, aggInt16Test) {
```

Comment:
rename `CiderNewAggTest` to `CiderNewAggHashTableTest`.

Reply:
Renamed.